src/main/java/sg/dex/starfish/impl/file/FileAsset.java

```diff
@@ -23,96 +21,135 @@
-			ob.put(CONTENT_TYPE, OCTET_STREAM);
-			ob.put(CONTENT_HASH, hashOfContent);
-
-			if (meta != null) {
-				Map<String,Object> metaMap = JSON.toMap(meta);
-				for (Map.Entry<String, Object> me : metaMap.entrySet()) {
-					ob.put(me.getKey(), me.getValue());
-				}
-			}
-
-			return ob;
-
-		} catch (FileNotFoundException e) {
-			throw new StarfishValidationException(e.getMessage());
-		}
-	}
-
-	/**
-	 * Gets an input stream that can be used to consume the content of this asset.
-	 *
-	 * Will throw an exception if consumption of the asset data in not possible
-	 * locally.
-	 * 
-	 * @throws AuthorizationException if requestor does not have access permission
-	 * @throws StorageException if unable to load the Asset
-	 * @return An input stream allowing consumption of the asset data
-	 */
-	@Override
-	public InputStream getContentStream() {
-		try {
-			return new FileInputStream(file);
-		}
-		catch (FileNotFoundException e) {
-			throw new StorageException("File not found ,file : "+file , e);
-		}
-	}
-
-	@Override
-	public long getContentSize() {
-		return null != file ? file.length(): -1;
-	}
-
+    private final File file;
+
+    protected FileAsset(String meta, File file) {
+        super(meta);
+        this.file = file;
+    }
+
+    /**
+     * This method is to crete Resource Asset with specific resource path, metadata  and isHashOfContentRequired
+     *
+     * @param f                       path of the resource
+     * @param isHashOfContentRequired if true then hash of content is calculated and included in metadata.
+     *                                Hash of content will be calculated using Keccak256 hash function
+     *                                if false then content hash is not included in metadata
+     * @return FileAsset instance created using given params
+     */
+    public static FileAsset create(File f, boolean isHashOfContentRequired) {
+        return new FileAsset(buildMetadata(f, null, isHashOfContentRequired), f);
+    }
+
+    /**
+     * This method is to crete Resource Asset with specific resource path, metadata  and isHashOfContentRequired
+     *
+     * @param f                       path of the file
+     * @param metaData                metadata associated with the asset.This metadata will be be added in addition to default
+     *                                metadata i.e DATE_CREATED,TYPE,CONTENT_TYPE.If same key,value is provided then the
+     *                                default value will be overridden.
+     * @param isHashOfContentRequired if true then hash of content is calculated and included in metadata.
+     *                                Hash of content will be calculated using Keccak256 hash function
+     *                                if false then content hash is not included in metadata
+     * @return FileAsset instance created using given params
+     */
+    public static FileAsset create(File f, String metaData, boolean isHashOfContentRequired) {
```

Comment:
This method signature doesn't make sense. If the user provides exact metadata, no need to do anything with a content hash.